### PR TITLE
[ISSUE-3046][Improve] Improve streampark-common module base on 3.3 Methods Rule

### LINKED ISSUE #3046: [Improve] Improve streampark-common module base on 3.3 Methods Rule
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Improve streampark-common module base on [3.3 Methods Rule](https://streampark.apache.org/community/submit_guide/code_style_and_quality_guide#33-methods-rule)

### Usage Scenario

_No response_

### Related issues

#2986

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengkezhou1**
## What changes were proposed in this pull request

Issue Number: close #3046 

## Brief change log

use **`@Nonnull`** and **`@Nullable`** in streampark-common/src/main/java/org/apache/streampark/common/
1. **enums/ApplicationType.java**
2. **ClusterState.java**
3. **enums/FlinkDevelopmentMode.java**
4. **enums/FlinkExecutionMode.java**
5. **enums/FlinkK8sRestExposedType.java**
6. **enums/FlinkRestoreMode.java**
7. **enums/FlinkSqlValidationFailedType.java**
8. **enums/ResolveOrder.java**
9. **enums/Semantic.java**
10. **enums/StorageType.java**
11. **util/ExceptionUtils.java**

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (no)

### CODE REVIEW

**Comment by caicancai** [comment]
Thank you for your contribution
cc @RocMarshal

File: streampark-common/src/main/java/org/apache/streampark/common/enums/ApplicationType.java [hunk 1]
```diff
@@ -43,10 +45,13 @@ public int getType() {
     return type;
   }
 
+  @Nonnull
   public String getName() {
     return name;
   }
 
+  /** switch param use this, can't be null */
+  @Nonnull
```
Comment (RocMarshal) [inline streampark-common/src/main/java/org/apache/streampark/common/enums/ApplicationType.java]:
```suggestion
  @Nullable
``` 
due to the line `61`

File: streampark-common/src/main/java/org/apache/streampark/common/enums/FlinkDevelopmentMode.java [hunk 2]
```diff
@@ -32,12 +35,14 @@ public enum FlinkDevelopmentMode {
...
 
-  FlinkDevelopmentMode(String name, Integer mode) {
+  FlinkDevelopmentMode(@Nonnull String name, @Nonnull Integer mode) {
     this.name = name;
     this.mode = mode;
   }
 
-  public static FlinkDevelopmentMode of(Integer value) {
+  /** switch param use this, can't be null */
+  @Nonnull
```
Comment (RocMarshal) [inline streampark-common/src/main/java/org/apache/streampark/common/enums/FlinkDevelopmentMode.java]: same as mentioned above

File: streampark-common/src/main/java/org/apache/streampark/common/enums/FlinkExecutionMode.java [hunk 3]
```diff
@@ -48,12 +51,14 @@ public enum FlinkExecutionMode {
...
 
-  FlinkExecutionMode(Integer mode, String name) {
+  FlinkExecutionMode(@Nonnull Integer mode, @Nonnull String name) {
     this.mode = mode;
     this.name = name;
   }
 
-  public static FlinkExecutionMode of(Integer value) {
+  /** switch param use this, can't be null */
+  @Nonnull
```
Comment (RocMarshal) [inline streampark-common/src/main/java/org/apache/streampark/common/enums/FlinkExecutionMode.java]: same as mentioned above

File: streampark-common/src/main/java/org/apache/streampark/common/enums/FlinkK8sRestExposedType.java [hunk 4]
```diff
@@ -33,12 +36,13 @@ public enum FlinkK8sRestExposedType {
...
   private final Integer type;
 
-  FlinkK8sRestExposedType(String name, Integer type) {
+  FlinkK8sRestExposedType(@Nonnull String name, @Nonnull Integer type) {
     this.name = name;
     this.type = type;
   }
 
-  public static FlinkK8sRestExposedType of(Integer value) {
+  @Nonnull
```
Comment (RocMarshal) [inline streampark-common/src/main/java/org/apache/streampark/common/enums/FlinkK8sRestExposedType.java]: same as mentioned above

File: streampark-common/src/main/java/org/apache/streampark/common/enums/FlinkRestoreMode.java [hunk 5]
```diff
@@ -55,11 +58,13 @@ public int get() {
     this.mode = mode;
   }
 
+  @Nonnull
   public String getName() {
     return this.toString();
   }
 
-  public static FlinkRestoreMode of(Integer value) {
+  @Nullable
```
Comment (RocMarshal) [inline streampark-common/src/main/java/org/apache/streampark/common/enums/FlinkRestoreMode.java:66]: 👍

File: streampark-common/src/main/java/org/apache/streampark/common/enums/ResolveOrder.java [hunk 6]
```diff
@@ -30,12 +33,13 @@ public enum ResolveOrder {
...
   private final Integer order;
 
-  ResolveOrder(String name, Integer order) {
+  ResolveOrder(@Nonnull String name, @Nonnull Integer order) {
     this.name = name;
     this.order = order;
   }
 
-  public static ResolveOrder of(Integer value) {
+  @Nullable
```
Comment (RocMarshal) [inline streampark-common/src/main/java/org/apache/streampark/common/enums/ResolveOrder.java:41]: 👍

File: streampark-common/src/main/java/org/apache/streampark/common/util/ExceptionUtils.java [hunk 7]
```diff
@@ -26,7 +29,8 @@ public class ExceptionUtils {
 
   private ExceptionUtils() {}
 
-  public static String stringifyException(Throwable throwable) {
+  @Nonnull
```
Comment (RocMarshal) [inline streampark-common/src/main/java/org/apache/streampark/common/util/ExceptionUtils.java:32]: 👍

**Review by RocMarshal — CHANGES_REQUESTED** [review]
Hi, @zzzk1 Thanks for your contribution~
Looks to me on the whole just left a few of comments, 
Please TALK in your free time~

**Comment by zhengkezhou1** [comment]
> Hi, @zzzk1 Thanks for your contribution~ Looks to me on the whole just left a few of comments, Please TALK in your free time~

 ok, i get it, but i found that these methods are involved in the **switch** keyword, which may cause a **NullPointerException**.

**Comment by zhengkezhou1** [comment]
hi, @RocMarshal  what should i do next ?

**Comment by caicancai** [comment]
> hi, @RocMarshal what should i do next ?

A small suggestion, usually if there is no reply within 24 hours after the change, @ reviewers

**Comment by wolfboys** [comment]
cc @RocMarshal PTAL again, thx

**Review by wolfboys — APPROVED** [review]
LGTM

**Comment by RocMarshal** [comment]
> ok, i get it, but i found that these methods are involved in the **switch** keyword, which may cause a **NullPointerException**.

Thanks @zzzk1  for the update and checking.
The current issue focuses only on improving under the rules. As we stated in the proposal, it is pure.
If you are interested the `NPE` risk, please create a new issue to describe and track the situation regarding switches and NPE, as it is a new issue. 
go for it !

**Review by RocMarshal — APPROVED** [review]
LGTM +1.
@zzzk1 thanks for your hard work!
merging~

**Pull request merged**